### ISSUE TITLE
fix(inference): forward_prefill_all_logits panics when LoRA is active (perplexity path)

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#1084]
## Evidence

`forward_prefill_all_logits` takes an explicit `panic!` when a LoRA adapter is active and all-position logits are requested (`crates/inference/src/forward/metal_qwen35.rs:6947-6954`):

```rust
if self.lora.is_some() {
    // Batched helper does not apply LoRA adapters; stay on sequential path.
    if all_positions {
        panic!(
            "forward_prefill_all_logits: LoRA active — \
             batch/all-position prefill does not apply LoRA"
        );
    }
    ...
```

The `all_positions` branch is reached by `compute_token_nlls` / perplexity computation.

## Impact

Requesting perplexity or per-token NLLs while a LoRA adapter is active panics the process. `panic!` in library code is banned by the crate conventions (`AGENTS.md`), and this is a valid-looking API combination (perplexity + adapter) rather than an impossible state, so it is a crash reachable from ordinary use.

## Fix shape

Return a typed error instead of panicking, so callers can handle the unsupported combination:

```rust
if all_positions {
    return Err(InferenceError::Inference(
        "all-position prefill does not apply LoRA adapters; use the sequential path".into(),
    ));
}
```

(Propagate the `Result` up through `forward_prefill_all_logits` and `compute_token_nlls` if the signature does not already return one.)

## Regression test

Call the perplexity/all-logits path with an active LoRA adapter: it must return `Err`, not panic. Must fail if the `panic!` is restored.
